src/android.rs

```diff
@@ -14,35 +14,44 @@ lazy_static::lazy_static! {
 }
 
 macro_rules! jni_handle_error {
-    ($env:expr, $err:ident) => {
-        if $env
-            .exception_check()
-            .map_err(|e| $crate::error::SecureEnvError::$err(e.to_string()))?
-        {
-            let throwable = $env
-                .exception_occurred()
-                .map_err(|e| $crate::error::SecureEnvError::$err(e.to_string()))?;
-            $env.exception_clear()
-                .map_err(|e| $crate::error::SecureEnvError::$err(e.to_string()))?;
-
-            let message = $env
-                .call_method(
-                    &throwable,
-                    EXCEPTION_TO_STRING,
-                    EXCEPTION_TO_STRING_SIG,
-                    &[],
-                )
-                .and_then(|v| v.l())
-                .map_err(|e| {
-                    $crate::error::SecureEnvError::UnableToCreateJavaValue(e.to_string())
-                })?;
-
-            let msg_rust: String = $env
-                .get_string(&message.into())
-                .map_err(|e| $crate::error::SecureEnvError::UnableToCreateJavaValue(e.to_string()))?
-                .into();
-
-            return Err($crate::error::SecureEnvError::$err(msg_rust));
+    ($env:expr, $err:ident, $e:expr) => {
+        match (|| -> $crate::error::SecureEnvResult<()> {
+            if $env
+                .exception_check()
+                .map_err(|e| $crate::error::SecureEnvError::$err(e.to_string()))?
+            {
+                let throwable = $env
+                    .exception_occurred()
+                    .map_err(|e| $crate::error::SecureEnvError::$err(e.to_string()))?;
+                $env.exception_clear()
+                    .map_err(|e| $crate::error::SecureEnvError::$err(e.to_string()))?;
+
+                let message = $env
+                    .call_method(
+                        &throwable,
+                        EXCEPTION_TO_STRING,
+                        EXCEPTION_TO_STRING_SIG,
+                        &[],
+                    )
+                    .and_then(|v| v.l())
+                    .map_err(|e| {
+                        $crate::error::SecureEnvError::UnableToCreateJavaValue(e.to_string())
+                    })?;
+
+                let msg_rust: String = $env
+                    .get_string(&message.into())
```

Comment:
Why would you not do the `into()` with the `message` variable but you do it with the `msg_rust` variable. This is inconsistent